src/KDSoapClient/KDSoapValue.cpp

```diff
@@ -367,6 +367,20 @@ QString KDSoapValue::type() const
     return d->m_typeName;
 }
 
+KDSoapValueList KDSoapValue::split() const
+{
+    KDSoapValueList valueList;
+    if (value().toString().trimmed().isEmpty()) return valueList;
+
+    const QStringList list = value().toString().split(QLatin1Char(' '));
+    for (int i = 0; i < list.count(); ++i) {
+        KDSoapValue value(*this);
+        value.setValue(list.at(i));
+        valueList << value;
```

Comment:
valueList.reserve(list.count()) should be done before the for loop, to minimize re-allocations.